api/src/main/java/io/strimzi/api/kafka/model/listener/KafkaListenerAuthenticationCustom.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.listener;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.strimzi.api.kafka.model.CertSecretSource;
+import io.strimzi.api.kafka.model.Constants;
+import io.strimzi.api.kafka.model.GenericSecretSource;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Configures a listener to use mutual TLS authentication.
+ */
+@Buildable(
+        editableEnabled = false,
+        builderPackage = Constants.FABRIC8_KUBERNETES_API
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@EqualsAndHashCode
+public class KafkaListenerAuthenticationCustom extends KafkaListenerAuthentication {
+
+    public static final String FORBIDDEN_PREFIXES = "ssl.keystore.location, ssl.keystore.password, ssl.keystore.type";
+
+    private static final long serialVersionUID = 1L;
+
+    public static final String TYPE_CUSTOM = "custom";
+
+    private Map<String, Object> listenerConfig;
+    private boolean sasl = true;
+    private List<CertSecretSource> tlsTrustedCertificates;
+    private List<GenericSecretSource> secrets;
+
+    @Description("Must be `" + TYPE_CUSTOM + "`")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @Override
+    public String getType() {
+        return TYPE_CUSTOM;
+    }
+
+    @Description("Enable or disable SASL on this listener. Default value is true.")
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    public boolean isSasl() {
+        return sasl;
```

Comment:
I wonder if this should be required instead of defaulting to true 🤔 . It would make it more obvious I guess.

Reply:
Yup agreed, think that works better by forcing the user to define it.